Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Sun Nov 26 11:40:05 2017 +0100

    Restore classic new tab page

diff --git a/chrome/browser/search/search.cc b/chrome/browser/search/search.cc
--- a/chrome/browser/search/search.cc
+++ b/chrome/browser/search/search.cc
@@ -115,10 +115,7 @@ bool IsURLAllowedForSupervisedUser(const GURL& url, Profile* profile) {
 }
 
 bool ShouldShowLocalNewTab(Profile* profile) {
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  return command_line->HasSwitch(switches::kForceLocalNtp) ||
-         (base::FeatureList::IsEnabled(features::kUseGoogleLocalNtp) &&
-          profile && DefaultSearchProviderIsGoogle(profile));
+  return true;
 }
 
 bool ShouldDelayRemoteNTP(const GURL& search_provider_url, Profile* profile) {
