Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Mon Feb 12 21:43:35 2018 +0100

    Never send any crash upload data

diff --git a/chrome/browser/tracing/crash_service_uploader.cc b/chrome/browser/tracing/crash_service_uploader.cc
--- a/chrome/browser/tracing/crash_service_uploader.cc
+++ b/chrome/browser/tracing/crash_service_uploader.cc
@@ -122,6 +122,9 @@ void TraceCrashServiceUploader::DoUpload(
     std::unique_ptr<const base::DictionaryValue> metadata,
     const UploadProgressCallback& progress_callback,
     const UploadDoneCallback& done_callback) {
+
+  return;
+
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
 
   progress_callback_ = progress_callback;
