From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Wed, 11 Jul 2018 11:17:53 +0200
Subject: Disable ads metrics by default

---
 chrome/common/chrome_features.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/common/chrome_features.cc b/chrome/common/chrome_features.cc
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -15,7 +15,7 @@ namespace features {
 // All features in alphabetical order.
 
 // Enables Ads Metrics.
-const base::Feature kAdsFeature{"AdsMetrics", base::FEATURE_ENABLED_BY_DEFAULT};
+const base::Feature kAdsFeature{"AdsMetrics", base::FEATURE_DISABLED_BY_DEFAULT};
 
 #if defined(OS_ANDROID)
 const base::Feature kAllowAutoplayUnmutedInWebappManifestScope{
-- 
2.7.4

