Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Sat Jan 20 19:44:43 2018 +0100

    Fix crash when accessing page info / site settings

diff --git a/chrome/browser/permissions/permission_manager.cc b/chrome/browser/permissions/permission_manager.cc
--- a/chrome/browser/permissions/permission_manager.cc
+++ b/chrome/browser/permissions/permission_manager.cc
@@ -601,6 +601,9 @@ PermissionResult PermissionManager::GetPermissionStatusHelper(
     const GURL& embedding_origin) {
   GURL canonical_requesting_origin = GetCanonicalOrigin(requesting_origin);
   PermissionContextBase* context = GetPermissionContext(permission);
+  if (context == nullptr) {
+      return PermissionResult(CONTENT_SETTING_BLOCK, PermissionStatusSource::UNSPECIFIED);
+  }
   PermissionResult result = context->GetPermissionStatus(
       render_frame_host, canonical_requesting_origin.GetOrigin(),
       embedding_origin.GetOrigin());
