From: samartnik <artem@brave.com>
Date: Thu, 14 Dec 2017 16:19:50 +0200
Subject: Always allow partner customisation

---
 .../PartnerBrowserCustomizations.java            | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/PartnerBrowserCustomizations.java b/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/PartnerBrowserCustomizations.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/PartnerBrowserCustomizations.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/PartnerBrowserCustomizations.java
@@ -324,14 +324,14 @@ public class PartnerBrowserCustomizations {
             @Override
             protected Void doInBackground() {
                 try {
-                    boolean systemOrPreStable =
-                            (context.getApplicationInfo().flags & ApplicationInfo.FLAG_SYSTEM) == 1
-                            || !ChromeVersionInfo.isStableBuild();
-                    if (!systemOrPreStable) {
-                        // Only allow partner customization if this browser is a system package, or
-                        // is in pre-stable channels.
-                        return null;
-                    }
+                    //boolean systemOrPreStable =
+                    //        (context.getApplicationInfo().flags & ApplicationInfo.FLAG_SYSTEM) == 1
+                    //        || !ChromeVersionInfo.isStableBuild();
+                    //if (!systemOrPreStable) {
+                    //    // Only allow partner customization if this browser is a system package, or
+                    //    // is in pre-stable channels.
+                    //    return null;
+                    //}
 
                     if (isCancelled()) return null;
                     Provider provider = AppHooks.get().getCustomizationProvider();
-- 
2.17.1

