From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 1 Dec 2020 00:29:28 -0500
Subject: ask permission to play protected media by default

---
 .../core/browser/content_settings_registry.cc            | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/components/content_settings/core/browser/content_settings_registry.cc b/components/content_settings/core/browser/content_settings_registry.cc
--- a/components/content_settings/core/browser/content_settings_registry.cc
+++ b/components/content_settings/core/browser/content_settings_registry.cc
@@ -78,16 +78,7 @@ std::set<ContentSetting> ValidSettings(ContentSetting setting1,
 }
 
 ContentSetting GetInitialDefaultContentSettingForProtectedMediaIdentifier() {
-// On Android, the default value is ALLOW or ASK depending on whether per-origin
-// provisioning is used (https://crbug.com/854737 and https://crbug.com/904883).
-// On ChromeOS the default value is always ASK.
-#if defined(OS_ANDROID)
-  return media::MediaDrmBridge::IsPerOriginProvisioningSupported()
-             ? CONTENT_SETTING_ALLOW
-             : CONTENT_SETTING_ASK;
-#else
   return CONTENT_SETTING_ASK;
-#endif  // defined(OS_ANDROID)
 }
 
 }  // namespace
-- 
2.17.1

