From b51c81df60e34bbd00f448b554a0f573ec934218 Mon Sep 17 00:00:00 2001
From: "guangxi.chen" <gxchen@nd.com.cn>
Date: Fri, 19 Nov 2021 02:01:39 +0000
Subject: [PATCH] change Android userAgent to deskmode userAgent

---
 content/common/user_agent.cc | 33 +++++++++++----------------------
 1 file changed, 11 insertions(+), 22 deletions(-)

diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
--- a/content/common/user_agent.cc
+++ b/content/common/user_agent.cc
@@ -36,7 +36,7 @@ std::string GetUserAgentPlatform() {
 #elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
   return "X11; ";  // strange, but that's what Firefox uses
 #elif BUILDFLAG(IS_ANDROID)
-  return "Linux; ";
+  return "X11; ";
 #elif BUILDFLAG(IS_FUCHSIA)
   return "";
 #else
@@ -47,9 +47,7 @@ std::string GetUserAgentPlatform() {
 }  // namespace

 std::string GetUnifiedPlatform() {
-#if BUILDFLAG(IS_ANDROID)
-  return frozen_user_agent_strings::kUnifiedPlatformAndroid;
-#elif BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_CHROMEOS)
   return frozen_user_agent_strings::kUnifiedPlatformCrOS;
 #elif BUILDFLAG(IS_MAC)
   return frozen_user_agent_strings::kUnifiedPlatformMacOS;
@@ -217,7 +215,7 @@ std::string BuildOSCpuInfoFromOSVersionAndCpuType(const std::string& os_version,
                                                   const std::string& cpu_type) {
   std::string os_cpu;

-#if !BUILDFLAG(IS_ANDROID) && BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_MAC)
+#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_MAC)
   // Should work on any Posix system.
   struct utsname unixinfo;
   uname(&unixinfo);
@@ -240,7 +238,9 @@ std::string BuildOSCpuInfoFromOSVersionAndCpuType(const std::string& os_version,
                       cpu_type.c_str(),  // e.g. i686
                       os_version.c_str()
 #elif BUILDFLAG(IS_ANDROID)
-                      "Android %s", os_version.c_str()
+                      "%s %s",
+                      unixinfo.sysname,  // e.g. Linux
+                      cpu_type.c_str()   // e.g. i686
 #elif BUILDFLAG(IS_FUCHSIA)
                       "Fuchsia"
 #elif BUILDFLAG(IS_POSIX)
@@ -256,20 +256,9 @@ std::string BuildOSCpuInfoFromOSVersionAndCpuType(const std::string& os_version,

 std::string GetReducedUserAgent(bool mobile, std::string major_version) {
   std::string user_agent;
-#if BUILDFLAG(IS_ANDROID)
-  std::string device_compat;
-  // Note: The extra space after Mobile is meaningful here, to avoid
-  // "MobileSafari", but unneeded for non-mobile Android devices.
-  device_compat = mobile ? "Mobile " : "";
-  user_agent = base::StringPrintf(frozen_user_agent_strings::kAndroid,
-                                  GetUnifiedPlatform().c_str(),
-                                  major_version.c_str(), device_compat.c_str());
-#else
   user_agent =
       base::StringPrintf(frozen_user_agent_strings::kDesktop,
                          GetUnifiedPlatform().c_str(), major_version.c_str());
-#endif
-
   return user_agent;
 }

@@ -292,10 +281,10 @@ std::string BuildUserAgentFromProductAndExtraOSInfo(
     const std::string& extra_os_info,
     IncludeAndroidBuildNumber include_android_build_number) {
   std::string os_info;
-  base::StrAppend(&os_info, {GetUserAgentPlatform(),
-                             BuildOSCpuInfo(include_android_build_number,
-                                            IncludeAndroidModel::Include),
-                             extra_os_info});
+  base::StringAppendF(&os_info, "%s%s", GetUserAgentPlatform().c_str(),
+                       BuildOSCpuInfo(IncludeAndroidBuildNumber::Exclude,
+                                      IncludeAndroidModel::Include)
+                           .c_str());
   return BuildUserAgentFromOSAndProduct(os_info, product);
 }

@@ -337,7 +326,7 @@ std::string BuildUserAgentFromOSAndProduct(const std::string& os_info,
   base::StringAppendF(&user_agent,
                       "Mozilla/5.0 (%s) AppleWebKit/537.36 (KHTML, like Gecko) "
                       "%s Safari/537.36",
-                      os_info.c_str(), product.c_str());
+                      os_info.c_str(), "");
   return user_agent;
 }

-- 
2.17.1

