From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 19 Dec 2020 20:04:14 +0100
Subject: Disable unified autoplay feature

---
 media/base/media_switches.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/base/media_switches.cc b/media/base/media_switches.cc
--- a/media/base/media_switches.cc
+++ b/media/base/media_switches.cc
@@ -498,7 +498,7 @@ const base::Feature kUseR16Texture{"use-r16-texture",
 // Enables the Unified Autoplay policy by overriding the platform's default
 // autoplay policy.
 const base::Feature kUnifiedAutoplay{"UnifiedAutoplay",
-                                     base::FEATURE_ENABLED_BY_DEFAULT};
+                                     base::FEATURE_DISABLED_BY_DEFAULT};
 
 #if defined(OS_LINUX)
 // Enable vaapi video decoding on linux. This is already enabled by default on
-- 
2.17.1

