From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Thu, 19 Aug 2021 23:11:16 +0200
Subject: Disable search engine DSE automatic permission

---
 components/permissions/features.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/permissions/features.cc b/components/permissions/features.cc
--- a/components/permissions/features.cc
+++ b/components/permissions/features.cc
@@ -76,7 +76,7 @@ const base::Feature kRevisedOriginHandling{"PermissionsRevisedOriginHandling",
 // When enabled, the Default Search Engine does not automatically receive the
 // "geolocation" and "notifications" permissions. DSE only applies to Android.
 const base::Feature kRevertDSEAutomaticPermissions{
-    "RevertDSEAutomaticPermissions", base::FEATURE_DISABLED_BY_DEFAULT};
+    "RevertDSEAutomaticPermissions", base::FEATURE_ENABLED_BY_DEFAULT};
 #endif  // defined(OS_ANDROID)
 
 }  // namespace features
-- 
2.17.1

