From: Zoraver Kang <zkang@wpi.edu>
Date: Mon, 26 Aug 2019 17:54:28 -0400
Subject: disable AdsBlockedInfoBar

---
 .../content/browser/profile_interaction_manager.cc | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/components/subresource_filter/content/browser/profile_interaction_manager.cc b/components/subresource_filter/content/browser/profile_interaction_manager.cc
--- a/components/subresource_filter/content/browser/profile_interaction_manager.cc
+++ b/components/subresource_filter/content/browser/profile_interaction_manager.cc
@@ -17,11 +17,6 @@
 #include "content/public/browser/render_frame_host.h"
 #include "content/public/browser/web_contents.h"
 
-#if defined(OS_ANDROID)
-#include "components/infobars/content/content_infobar_manager.h"  // nogncheck
-#include "components/subresource_filter/content/browser/ads_blocked_infobar_delegate.h"
-#endif
-
 namespace subresource_filter {
 
 ProfileInteractionManager::ProfileInteractionManager(
@@ -135,15 +130,6 @@ void ProfileInteractionManager::MaybeShowNotification() {
   const GURL& top_level_url = page_->GetMainDocument().GetLastCommittedURL();
   if (profile_context_->settings_manager()->ShouldShowUIForSite(
           top_level_url)) {
-#if defined(OS_ANDROID)
-    // NOTE: It is acceptable for the embedder to not have installed an infobar
-    // manager.
-    if (auto* infobar_manager =
-            infobars::ContentInfoBarManager::FromWebContents(
-                GetWebContents())) {
-      subresource_filter::AdsBlockedInfobarDelegate::Create(infobar_manager);
-    }
-#endif
 
     // TODO(https://crbug.com/1103176): Plumb the actual frame reference here
     // (it comes from
-- 
2.17.1

