From: Zoraver Kang <zkang@wpi.edu>
Date: Mon, 26 Aug 2019 17:54:28 -0400
Subject: disable AdsBlockedInfoBar

---
 .../chrome_subresource_filter_client.cc                | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc b/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
--- a/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
+++ b/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
@@ -6,7 +6,6 @@
 
 #include "build/build_config.h"
 #include "chrome/browser/browser_process.h"
-#include "chrome/browser/infobars/infobar_service.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/safe_browsing/safe_browsing_service.h"
 #include "chrome/browser/subresource_filter/subresource_filter_profile_context_factory.h"
@@ -14,10 +13,6 @@
 #include "components/subresource_filter/content/browser/content_subresource_filter_throttle_manager.h"
 #include "components/subresource_filter/content/browser/ruleset_service.h"
 
-#if defined(OS_ANDROID)
-#include "components/subresource_filter/android/ads_blocked_infobar_delegate.h"
-#endif
-
 ChromeSubresourceFilterClient::ChromeSubresourceFilterClient(
     content::WebContents* web_contents)
     : web_contents_(web_contents) {
@@ -44,11 +39,6 @@ void ChromeSubresourceFilterClient::
 }
 
 void ChromeSubresourceFilterClient::ShowNotification() {
-#if defined(OS_ANDROID)
-    InfoBarService* infobar_service =
-        InfoBarService::FromWebContents(web_contents_);
-    subresource_filter::AdsBlockedInfobarDelegate::Create(infobar_service);
-#endif
 }
 
 const scoped_refptr<safe_browsing::SafeBrowsingDatabaseManager>
-- 
2.17.1

