From: Zoraver Kang <zkang@wpi.edu>
Date: Mon, 26 Aug 2019 17:54:28 -0400
Subject: disable AdsBlockedInfoBar

---
 .../chrome_subresource_filter_client.cc                | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc b/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
--- a/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
+++ b/chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
@@ -11,7 +11,6 @@
 #include "base/metrics/histogram_macros.h"
 #include "build/build_config.h"
 #include "chrome/browser/browser_process.h"
-#include "chrome/browser/infobars/infobar_service.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/safe_browsing/safe_browsing_service.h"
 #include "chrome/browser/subresource_filter/subresource_filter_profile_context_factory.h"
@@ -33,10 +32,6 @@
 #include "content/public/browser/web_contents.h"
 #include "services/metrics/public/cpp/ukm_source_id.h"
 
-#if defined(OS_ANDROID)
-#include "components/subresource_filter/android/ads_blocked_infobar_delegate.h"
-#endif
-
 ChromeSubresourceFilterClient::ChromeSubresourceFilterClient(
     content::WebContents* web_contents)
     : web_contents_(web_contents) {
@@ -126,11 +121,6 @@ ChromeSubresourceFilterClient::GetProfileInteractionManager() {
 }
 
 void ChromeSubresourceFilterClient::ShowUI(const GURL& url) {
-#if defined(OS_ANDROID)
-  InfoBarService* infobar_service =
-      InfoBarService::FromWebContents(web_contents_);
-  subresource_filter::AdsBlockedInfobarDelegate::Create(infobar_service);
-#endif
   // TODO(https://crbug.com/1103176): Plumb the actual frame reference here
   // (it comes  from
   // ContentSubresourceFilterThrottleManager::DidDisallowFirstSubresource, which
-- 
2.17.1

