From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Mon, 12 Aug 2019 08:45:43 +0200
Subject: Disable unified consent on Android

---
 components/unified_consent/feature.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/unified_consent/feature.cc b/components/unified_consent/feature.cc
--- a/components/unified_consent/feature.cc
+++ b/components/unified_consent/feature.cc
@@ -11,7 +11,7 @@ namespace unified_consent {
 // base::Feature definition.
 const base::Feature kUnifiedConsent {
   "UnifiedConsent",
-#if defined(OS_CHROMEOS)
+#if defined(OS_CHROMEOS) || defined(OS_ANDROID)
       base::FEATURE_DISABLED_BY_DEFAULT
 #else
       base::FEATURE_ENABLED_BY_DEFAULT
-- 
2.11.0

