From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 27 Oct 2018 17:54:01 +0200
Subject: Disable update scheduler

---
 .../components/component_updater/UpdateScheduler.java         | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/components/component_updater/android/java/src/org/chromium/components/component_updater/UpdateScheduler.java b/components/component_updater/android/java/src/org/chromium/components/component_updater/UpdateScheduler.java
--- a/components/component_updater/android/java/src/org/chromium/components/component_updater/UpdateScheduler.java
+++ b/components/component_updater/android/java/src/org/chromium/components/component_updater/UpdateScheduler.java
@@ -34,9 +34,7 @@ public class UpdateScheduler {
 
     @CalledByNative
     /* package */ static boolean isAvailable() {
-        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
-                || ChromiumPlayServicesAvailability.isGooglePlayServicesAvailable(
-                        ContextUtils.getApplicationContext());
+        return false;
     }
 
     /* package */ void onStartTaskBeforeNativeLoaded(TaskFinishedCallback callback) {
-- 
2.17.1

