From: Daniel Micay <danielmicay@gmail.com>
Date: Sat, 26 Nov 2016 14:57:22 -0500
Subject: Disable hyperlink auditing by default

---
 chrome/browser/about_flags.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1325,7 +1325,7 @@ const FeatureEntry kFeatureEntries[] = {
      SINGLE_VALUE_TYPE(switches::kDisableIpcFloodingProtection)},
     {"disable-hyperlink-auditing", flag_descriptions::kHyperlinkAuditingName,
      flag_descriptions::kHyperlinkAuditingDescription, kOsAll,
-     SINGLE_DISABLE_VALUE_TYPE(switches::kNoPings)},
+     SINGLE_VALUE_TYPE(switches::kNoPings)},
 #if defined(OS_ANDROID)
     {"contextual-search", flag_descriptions::kContextualSearchName,
      flag_descriptions::kContextualSearchDescription, kOsAndroid,
-- 
2.11.0

