From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 27 Oct 2017 15:53:37 +0200
Subject: Disable data saver

---
 .../datareduction/DataReductionPreferenceFragment.java         | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPreferenceFragment.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPreferenceFragment.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPreferenceFragment.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPreferenceFragment.java
@@ -48,10 +48,10 @@ public class DataReductionPreferenceFragment extends PreferenceFragment {
 
         PreferenceUtils.addPreferencesFromResource(this, R.xml.data_reduction_preferences);
         getActivity().setTitle(R.string.data_reduction_title_lite_mode);
-        boolean isEnabled = DataReductionProxySettings.getInstance().isDataReductionProxyEnabled();
-        mIsEnabled = !isEnabled;
-        mWasEnabledAtCreation = isEnabled;
-        updatePreferences(isEnabled);
+
+        mIsEnabled = false;
+        mWasEnabledAtCreation = false;
+        updatePreferences(false);
 
         setHasOptionsMenu(true);
 
@@ -148,6 +148,7 @@ public class DataReductionPreferenceFragment extends PreferenceFragment {
     }
 
     private void createDataReductionSwitch(boolean isEnabled) {
+        if (false) {
         final ChromeSwitchPreference dataReductionSwitch =
                 new ChromeSwitchPreference(getActivity(), null);
         dataReductionSwitch.setKey(PREF_DATA_REDUCTION_SWITCH);
@@ -175,5 +176,6 @@ public class DataReductionPreferenceFragment extends PreferenceFragment {
         // (e.g. the switch will say "On" when data reduction is really turned off), so
         // .setChecked() should be called after .addPreference()
         dataReductionSwitch.setChecked(isEnabled);
+        }
     }
 }
-- 
2.11.0

