From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 11 Jul 2020 15:00:13 +0200
Subject: do not add suffix to package name

---
 chrome/android/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -45,7 +45,7 @@ if (android_64bit_target_cpu && skip_secondary_abi_for_cq) {
 }
 
 _default_package = "org.chromium.chrome"
-if (android_channel != "default") {
+if (android_channel != "default" && android_channel != "stable") {
   # android-binary-size trybot may checks if an internal Chrome variant's
   # AndroidManifest is as expected by ensuring the differences between its
   # AndroidManifest and its upstream target's .AndroidManifest.expected file
--
2.25.1
