From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 27 Jul 2019 11:19:53 +0200
Subject: Do not record UMA for network errors

---
 services/network/network_context.cc | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/services/network/network_context.cc b/services/network/network_context.cc
--- a/services/network/network_context.cc
+++ b/services/network/network_context.cc
@@ -393,23 +393,6 @@ class NetworkContext::ContextNetworkDelegate
                                                                  started);
     }
 
-    // Record network errors that HTTP requests complete with, including OK and
-    // ABORTED.
-    // TODO(mmenke): Seems like this really should be looking at HTTPS requests,
-    // too.
-    // TODO(mmenke): We should remove the main frame case from here, and move it
-    // into the consumer - the network service shouldn't know what a main frame
-    // is.
-    if (request->url().SchemeIs("http")) {
-      base::UmaHistogramSparse("Net.HttpRequestCompletionErrorCodes",
-                               -net_error);
-
-      if (request->load_flags() & net::LOAD_MAIN_FRAME_DEPRECATED) {
-        base::UmaHistogramSparse(
-            "Net.HttpRequestCompletionErrorCodes.MainFrame", -net_error);
-      }
-    }
-
     ForwardProxyErrors(net_error);
   }
 
-- 
2.11.0

