From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 17 Mar 2019 19:09:00 +0100
Subject: Enable site per process isolation for devices with enough memory

---
 chrome/common/chrome_features.cc | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/chrome/common/chrome_features.cc b/chrome/common/chrome_features.cc
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -655,11 +655,7 @@ const base::Feature kShowTrustedPublisherURL{"ShowTrustedPublisherURL",
 // browser_features, as they are only used on the browser side.
 const base::Feature kSitePerProcess {
   "site-per-process",
-#if defined(OS_ANDROID)
-      base::FEATURE_DISABLED_BY_DEFAULT
-#else
       base::FEATURE_ENABLED_BY_DEFAULT
-#endif
 };
 
 // Controls a mode for dynamically process-isolating sites where the user has
@@ -689,7 +685,7 @@ const base::Feature kSiteIsolationForPasswordSites {
 // base::SysInfo::AmountOfPhysicalMemoryMB().
 const base::Feature kSitePerProcessOnlyForHighMemoryClients{
     "site-per-process-only-for-high-memory-clients",
-    base::FEATURE_DISABLED_BY_DEFAULT};
+    base::FEATURE_ENABLED_BY_DEFAULT};
 const char kSitePerProcessOnlyForHighMemoryClientsParamName[] =
     "site-per-process-low-memory-cutoff-mb";
 
-- 
2.17.1

