From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 20 Jan 2018 19:44:43 +0100
Subject: Fix crash when accessing page info / site settings

---
 chrome/browser/permissions/permission_manager.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/chrome/browser/permissions/permission_manager.cc b/chrome/browser/permissions/permission_manager.cc
--- a/chrome/browser/permissions/permission_manager.cc
+++ b/chrome/browser/permissions/permission_manager.cc
@@ -784,6 +784,9 @@ permissions::PermissionResult PermissionManager::GetPermissionStatusHelper(
     return permissions::PermissionResult(
         status, permissions::PermissionStatusSource::UNSPECIFIED);
   PermissionContextBase* context = GetPermissionContext(permission);
+  if (context == nullptr) {
+      return permissions::PermissionResult(CONTENT_SETTING_BLOCK, permissions::PermissionStatusSource::UNSPECIFIED);
+  }
   permissions::PermissionResult result = context->GetPermissionStatus(
       render_frame_host, canonical_requesting_origin.GetOrigin(),
       embedding_origin.GetOrigin());
-- 
2.17.1

