From: uazo <uazo@users.noreply.github.com>
Date: Tue, 11 Jan 2022 13:57:10 +0000
Subject: Fix startup crash from v97

---
 .../chrome/browser/init/StartupTabPreloader.java   | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java b/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
@@ -141,20 +141,6 @@ public class StartupTabPreloader implements ProfileManager.Observer, DestroyObse
     public void onFirstNavigationCommit() {
         recordDurationFromLoadDecisionIntoHistogram(
                 "Android.StartupTabPreloader.LoadDecisionToFirstNavigationCommit");
-
-        // We record the metric for navigation start here as well, as we want that metric to be
-        // recorded only for navigations that result in the first navigation commit startup metric
-        // being recorded.
-        if (mLoadDecisionMs == 0) return;
-        assert mFirstNavigationStartMs > 0;
-        long triggerpointToFirstNavigationStartMs = mFirstNavigationStartMs - mLoadDecisionMs;
-
-        // Note that we don't use recordDurationFromLoadDecisionIntoHistogram() here as first
-        // navigation start was reached before tab matching occurred.
-        String suffix = preloadWasViable() ? ".Load" : ".NoLoad";
-        RecordHistogram.recordMediumTimesHistogram(
-                "Android.StartupTabPreloader.LoadDecisionToFirstNavigationStart" + suffix,
-                triggerpointToFirstNavigationStartMs);
     }
 
     @Override
-- 
2.20.1

