From: uazo <uazo@users.noreply.github.com>
Date: Tue, 11 Jan 2022 13:57:10 +0000
Subject: Fix startup crash from v97

---
 .../chrome/browser/init/StartupTabPreloader.java       | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java b/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/init/StartupTabPreloader.java
@@ -170,16 +170,6 @@ public class StartupTabPreloader implements ProfileManager.Observer, DestroyObse
 
     @Override
     public void onFirstNavigationCommit() {
-        recordDurationFromLoadDecisionIntoPostTabMatchHistogram(
-                "Android.StartupTabPreloader.LoadDecisionToFirstNavigationCommit");
-
-        // We record the metric for navigation start here as well, as we want that metric to be
-        // recorded only for navigations that result in the first navigation commit startup metric
-        // being recorded.
-        assert mFirstNavigationStartMs > 0;
-        recordDurationFromLoadDecisionToEventTimeIntoPreTabMatchHistogram(
-                "Android.StartupTabPreloader.LoadDecisionToFirstNavigationStart",
-                mFirstNavigationStartMs);
     }
 
     @Override
-- 
2.25.1

