From: krlvm <51774833+krlvm@users.noreply.github.com>
Date: Mon, 4 Jul 2022 16:14:37 +0300
Subject: Follow only system dark mode preference when theme is set to system default

---
 GlobalNightModeStateController.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/GlobalNightModeStateController.java b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/GlobalNightModeStateController.java
index 726e7f1..aedb911 100644
--- a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/GlobalNightModeStateController.java
+++ b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/GlobalNightModeStateController.java
@@ -126,7 +126,7 @@ class GlobalNightModeStateController implements NightModeStateProvider,
     }
 
     private void updateNightMode() {
-        boolean powerSaveModeOn = mPowerSaveModeMonitor.powerSavingIsOn();
+        boolean powerSaveModeOn = false;
         final int theme = NightModeUtils.getThemeSetting();
         final boolean newNightModeOn = theme == ThemeType.SYSTEM_DEFAULT
                         && (powerSaveModeOn || mSystemNightModeMonitor.isSystemNightModeOn())
-- 
2.25.1

