From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 5 Sep 2020 21:38:15 +0200
Subject: Hardening against incognito mode detection

FILE:Hardening-against-incognito-mode-detection.patch
---
diff --git a/storage/browser/quota/quota_settings.cc b/storage/browser/quota/quota_settings.cc
index 4b8ac29b31..9cc423d302 100644
--- a/storage/browser/quota/quota_settings.cc
+++ b/storage/browser/quota/quota_settings.cc
@@ -29,7 +29,7 @@ const int kRandomizedPercentage = 10;
 const double kDefaultPerHostRatio = 0.75;
 const double kDefaultPoolSizeRatio = 0.8;
 const double kIncognitoQuotaRatioLowerBound = 0.15;
-const double kIncognitoQuotaRatioUpperBound = 0.2;
+const double kIncognitoQuotaRatioUpperBound = 0.3;
 
 // Skews |value| by +/- |percent|.
 int64_t RandomizeByPercent(int64_t value, int percent) {
