From beeda0bb81f59cfe272769eb393b0c58c5283ae9 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Mon, 24 Jan 2022 19:22:26 +0000
Subject: [PATCH] Hide currently no-op preload settings

---
 .../chrome/browser/privacy/settings/PrivacySettings.java         | 1 +
 1 file changed, 1 insertion(+)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
index 67a8c11ace8e0..9dd54398f36ba 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
@@ -131,6 +131,7 @@ public class PrivacySettings
         Preference preloadPagesPreference = findPreference(PREF_PRELOAD_PAGES);
         preloadPagesPreference.setSummary(
                 PreloadPagesSettingsFragment.getPreloadPagesSummaryString(getContext()));
+        getPreferenceScreen().removePreference(preloadPagesPreference);
 
         ChromeSwitchPreference httpsFirstModePref =
                 (ChromeSwitchPreference) findPreference(PREF_HTTPS_FIRST_MODE);
