From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 15 Dec 2019 17:59:04 +0100
Subject: Log DoH timeout

---
 net/dns/dns_session.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/net/dns/dns_session.cc b/net/dns/dns_session.cc
--- a/net/dns/dns_session.cc
+++ b/net/dns/dns_session.cc
@@ -325,9 +325,11 @@ base::TimeDelta DnsSession::NextTimeout(unsigned server_index, int attempt) {
 }
 
 base::TimeDelta DnsSession::NextDohTimeout(unsigned doh_server_index) {
-  return NextTimeoutHelper(
+  base::TimeDelta delta = NextTimeoutHelper(
       GetServerStats(doh_server_index, true /* is _doh_server */),
       0 /* num_backoffs */);
+  LOG(INFO) << "DoH: using timeout: " << delta;
+  return delta;
 }
 
 base::TimeDelta DnsSession::NextTimeoutHelper(ServerStats* server_stats,
-- 
2.17.1

