From: AlexeyBarabash <alexey@brave.com>
Date: Thu, 11 Jan 2018 13:27:34 +0200
Subject: Open YouTube links in Bromite

---
 .../external_intents/ExternalNavigationHandler.java       | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
--- a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
+++ b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
@@ -493,6 +493,7 @@ public class ExternalNavigationHandler {
             startActivity(intent, false, mDelegate);
             return OverrideUrlLoadingResult.OVERRIDE_WITH_EXTERNAL_INTENT;
         }
+
     }
 
     private static void loadUrlWithReferrer(
@@ -1060,6 +1061,13 @@ public class ExternalNavigationHandler {
             return OverrideUrlLoadingResult.OVERRIDE_WITH_ASYNC_ACTION;
         }
 
+        // force to open YouTube urls in Bromite
+        String intentPackageName = targetIntent.getPackage();
+        if (intentPackageName != null && intentPackageName.equals("com.google.android.youtube")) {
+            if (DEBUG) Log.i(TAG, "NO_OVERRIDE: YouTube URL for YouTube app");
+            return OverrideUrlLoadingResult.NO_OVERRIDE;
+        }
+
         // This should come after file intents, but before any returns of
         // OVERRIDE_WITH_EXTERNAL_INTENT.
         if (externalIntentRequestsDisabledForUrl(params)) {
-- 
2.17.1

