From: AlexeyBarabash <alexey@brave.com>
Date: Thu, 11 Jan 2018 13:27:34 +0200
Subject: Open YouTube links in Bromite

---
 .../chrome/browser/externalnav/ExternalNavigationHandler.java      | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java b/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
@@ -400,6 +400,13 @@ public class ExternalNavigationHandler {
         }
 
         // TODO(changwan): check if we need to handle URL even when external intent is off.
+        // Force to open YouTube urls in Bromite
+        String intentPackageName = intent.getPackage();
+        if (intentPackageName != null && intentPackageName.equals("com.google.android.youtube")) {
+            if (DEBUG) Log.i(TAG, "NO_OVERRIDE: YouTube URL for YouTube app");
+            return OverrideUrlLoadingResult.NO_OVERRIDE;
+        }
+
         if (CommandLine.getInstance().hasSwitch(
                 ChromeSwitches.DISABLE_EXTERNAL_INTENT_REQUESTS)) {
             Log.w(TAG, "External intent handling is disabled by a command-line flag.");
-- 
2.11.0

