From: uazo <uazo@users.noreply.github.com>
Date: Wed, 19 Jan 2022 12:39:24 +0000
Subject: Remove fetching google fonts from gms

---
 .../chromium/content/browser/font/AndroidFontLookupImpl.java    | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/content/public/android/java/src/org/chromium/content/browser/font/AndroidFontLookupImpl.java b/content/public/android/java/src/org/chromium/content/browser/font/AndroidFontLookupImpl.java
--- a/content/public/android/java/src/org/chromium/content/browser/font/AndroidFontLookupImpl.java
+++ b/content/public/android/java/src/org/chromium/content/browser/font/AndroidFontLookupImpl.java
@@ -233,6 +233,8 @@ public class AndroidFontLookupImpl implements AndroidFontLookup {
      * @return An opened font file descriptor, or null if the font file is not available.
      */
     private ParcelFileDescriptor tryFetchFont(String fontUniqueName) {
+        if ((true)) return null;
+
         String query = mFullFontNameToQuery.get(fontUniqueName);
         if (query == null) {
             Log.d(TAG, "Query format not found for full font name: %s", fontUniqueName);
-- 
2.20.1

