From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 21 Aug 2020 22:39:23 +0200
Subject: Remove blocklisted URLs upon bookmark creation

---
 .../chrome/browser/app/ChromeActivity.java      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
@@ -138,6 +138,8 @@ import org.chromium.chrome.browser.metrics.LaunchMetrics;
 import org.chromium.chrome.browser.metrics.UmaSessionStats;
 import org.chromium.chrome.browser.multiwindow.MultiWindowUtils;
 import org.chromium.chrome.browser.ntp.NewTabPageUma;
+import org.chromium.chrome.browser.suggestions.SuggestionsDependencyFactory;
+import org.chromium.chrome.browser.suggestions.mostvisited.MostVisitedSites;
 import org.chromium.chrome.browser.offlinepages.OfflinePageUtils;
 import org.chromium.chrome.browser.offlinepages.indicator.OfflineIndicatorController;
 import org.chromium.chrome.browser.omaha.UpdateInfoBarController;
@@ -238,6 +240,8 @@ import org.chromium.ui.widget.Toast;
 import org.chromium.url.GURL;
 import org.chromium.webapk.lib.client.WebApkNavigationClient;
 
+import org.chromium.url.GURL;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Consumer;
@@ -287,6 +291,7 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
     private UmaSessionStats mUmaSessionStats;
     private ContextReporter mContextReporter;
+    private MostVisitedSites mMostVisitedSites;
 
     private boolean mPartnerBrowserRefreshNeeded;
 
@@ -1439,6 +1444,11 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
             mCompositorViewHolderSupplier.set(null);
         }
 
+        if (mMostVisitedSites != null) {
+                mMostVisitedSites.destroy();
+                mMostVisitedSites = null;
+        }
+
         onDestroyInternal();
 
         if (mDidAddPolicyChangeListener) {
@@ -1749,6 +1759,13 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
         // Defense in depth against the UI being erroneously enabled.
         BookmarkBridge bridge = mBookmarkBridgeSupplier.get();
+	// remove blocklisted URL from most visited sites
+	if (mMostVisitedSites == null) {
+            mMostVisitedSites =
+                SuggestionsDependencyFactory.getInstance().createMostVisitedSites(Profile.getLastUsedRegularProfile());
+        }
+        mMostVisitedSites.removeBlocklistedUrl(tabToBookmark.getOriginalUrl());	
+
         if (bridge == null || !bridge.isEditBookmarksEnabled()) {
             assert false;
             return;
-- 
2.17.1

