From: Wengling Chen <feiyu2817@gmail.com>
Date: Sat, 2 Nov 2019 09:29:52 +0100
Subject: Remove dependency on com.google.android.gcm

---
 third_party/android_sdk/BUILD.gn | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/third_party/android_sdk/BUILD.gn b/third_party/android_sdk/BUILD.gn
--- a/third_party/android_sdk/BUILD.gn
+++ b/third_party/android_sdk/BUILD.gn
@@ -63,7 +63,4 @@ if (enable_java_templates) {
     testonly = true
     sources = [ "//third_party/android_sdk/public/extras/chromium/support/src/org/chromium/android/support/PackageManagerWrapper.java" ]
   }
-  android_java_prebuilt("android_gcm_java") {
-    jar_path = "//third_party/android_sdk/public/extras/google/gcm/gcm-client/dist/gcm.jar"
-  }
 }
-- 
2.17.1

