From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 3 May 2020 12:15:33 +0200
Subject: Remove google_services_gcm

---
 third_party/android_deps/BUILD.gn | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -995,24 +995,6 @@ java_prebuilt("com_android_tools_build_jetifier_jetifier_processor_java") {
   ]
 }
 
-# This is generated, do not edit. Update BuildConfigGenerator.groovy instead.
-android_aar_prebuilt("google_play_services_gcm_java") {
-  aar_path = "libs/com_google_android_gms_play_services_gcm/play-services-gcm-17.0.0.aar"
-  info_path = "libs/com_google_android_gms_play_services_gcm/com_google_android_gms_play_services_gcm.info"
-  deps = [
-    ":androidx_collection_collection_java",
-    ":androidx_core_core_java",
-    ":androidx_legacy_legacy_support_core_utils_java",
-    ":google_play_services_base_java",
-    ":google_play_services_basement_java",
-    ":google_play_services_iid_java",
-    ":google_play_services_stats_java",
-  ]
-
-  # Removing drawables from GMS .aars as they are unused bloat.
-  strip_drawables = true
-}
-
 # This is generated, do not edit. Update BuildConfigGenerator.groovy instead.
 android_aar_prebuilt("com_google_android_material_material_java") {
   aar_path =
-- 
2.17.1

