From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 21 Aug 2020 22:39:23 +0200
Subject: Remove blocklisted URLs upon bookmark creation

---
 .../chrome/browser/app/ChromeActivity.java      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
@@ -141,6 +141,8 @@ import org.chromium.chrome.browser.settings.SettingsLauncherImpl;
 import org.chromium.chrome.browser.settings.AdBlockPreferences;
 import org.chromium.chrome.browser.share.ShareDelegate;
 import org.chromium.chrome.browser.share.ShareDelegateImpl;
+import org.chromium.chrome.browser.suggestions.SuggestionsDependencyFactory;
+import org.chromium.chrome.browser.suggestions.mostvisited.MostVisitedSites;
 import org.chromium.chrome.browser.tab.AccessibilityVisibilityHandler;
 import org.chromium.chrome.browser.tab.Tab;
 import org.chromium.chrome.browser.tab.TabHidingType;
@@ -212,6 +214,8 @@ import org.chromium.ui.widget.Toast;
 import org.chromium.url.Origin;
 import org.chromium.webapk.lib.client.WebApkNavigationClient;
 
+import org.chromium.url.GURL;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Consumer;
@@ -256,6 +260,7 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
     private UmaSessionStats mUmaSessionStats;
     private ContextReporter mContextReporter;
+    private MostVisitedSites mMostVisitedSites;
 
     private boolean mPartnerBrowserRefreshNeeded;
 
@@ -1255,6 +1260,11 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
             mCompositorViewHolder = null;
         }
 
+        if (mMostVisitedSites != null) {
+                mMostVisitedSites.destroy();
+                mMostVisitedSites = null;
+        }
+
         onDestroyInternal();
 
         if (mDidAddPolicyChangeListener) {
@@ -1491,6 +1501,13 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
         // Defense in depth against the UI being erroneously enabled.
         BookmarkBridge bridge = mBookmarkBridgeSupplier.get();
+	// remove blocklisted URL from most visited sites
+	if (mMostVisitedSites == null) {
+            mMostVisitedSites =
+                SuggestionsDependencyFactory.getInstance().createMostVisitedSites(Profile.getLastUsedRegularProfile());
+        }
+        mMostVisitedSites.removeBlocklistedUrl(new GURL(tabToBookmark.getOriginalUrl()));
+
         if (bridge == null || !bridge.isEditBookmarksEnabled()) {
             assert false;
             return;
-- 
2.17.1

