From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 27 Sep 2020 09:24:16 +0200
Subject: Remove weblayer dependency on Play Services

FILE:Remove-weblayer-dependency-on-Play-Services.patch
---
diff --git a/weblayer/shell/android/BUILD.gn b/weblayer/shell/android/BUILD.gn
index 6f9c0eccf9..07d0ac9bf1 100644
--- a/weblayer/shell/android/BUILD.gn
+++ b/weblayer/shell/android/BUILD.gn
@@ -36,8 +36,6 @@ android_library("weblayer_shell_java") {
   testonly = true
   resources_package = "org.chromium.weblayer.shell"
 
-  # The google_play_services_base_java dependency is necessary for the
-  # google_play_services_version definition in the manifest.
   deps = [
     ":weblayer_shell_resources",
     "//base:base_java",
