From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 21 Aug 2020 22:39:23 +0200
Subject: Remove blocklisted URLs upon bookmark creation

FILE:Remove-blocklisted-URLs-upon-bookmark-creation.patch
---
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
index 13d1283941..d2c8788bea 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
@@ -143,6 +143,8 @@ import org.chromium.chrome.browser.settings.SettingsLauncherImpl;
 import org.chromium.chrome.browser.settings.AdBlockPreferences;
 import org.chromium.chrome.browser.share.ShareDelegate;
 import org.chromium.chrome.browser.share.ShareDelegateImpl;
+import org.chromium.chrome.browser.suggestions.SuggestionsDependencyFactory;
+import org.chromium.chrome.browser.suggestions.mostvisited.MostVisitedSites;
 import org.chromium.chrome.browser.tab.AccessibilityVisibilityHandler;
 import org.chromium.chrome.browser.tab.Tab;
 import org.chromium.chrome.browser.tab.TabHidingType;
@@ -216,6 +218,8 @@ import org.chromium.url.GURL;
 import org.chromium.url.Origin;
 import org.chromium.webapk.lib.client.WebApkNavigationClient;
 
+import org.chromium.url.GURL;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Consumer;
@@ -260,6 +264,7 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
     private UmaSessionStats mUmaSessionStats;
     private ContextReporter mContextReporter;
+    private MostVisitedSites mMostVisitedSites;
 
     private boolean mPartnerBrowserRefreshNeeded;
 
@@ -1273,6 +1278,11 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
             mCompositorViewHolder = null;
         }
 
+        if (mMostVisitedSites != null) {
+                mMostVisitedSites.destroy();
+                mMostVisitedSites = null;
+        }
+
         onDestroyInternal();
 
         if (mDidAddPolicyChangeListener) {
@@ -1521,6 +1531,13 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
 
         // Defense in depth against the UI being erroneously enabled.
         BookmarkBridge bridge = mBookmarkBridgeSupplier.get();
+	// remove blocklisted URL from most visited sites
+	if (mMostVisitedSites == null) {
+            mMostVisitedSites =
+                SuggestionsDependencyFactory.getInstance().createMostVisitedSites(Profile.getLastUsedRegularProfile());
+        }
+        mMostVisitedSites.removeBlocklistedUrl(new GURL(tabToBookmark.getOriginalUrl()));
+
         if (bridge == null || !bridge.isEditBookmarksEnabled()) {
             assert false;
             return;
