From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 27 Oct 2019 10:00:13 +0100
Subject: Replace DoH probe domain with RIPE domain

---
 net/dns/dns_transaction.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/dns/dns_transaction.cc b/net/dns/dns_transaction.cc
--- a/net/dns/dns_transaction.cc
+++ b/net/dns/dns_transaction.cc
@@ -838,7 +838,7 @@ class DnsTCPAttempt : public DnsAttempt {
 
 // ----------------------------------------------------------------------------
 
-const char kDoHProbeHostname[] = "www.gstatic.com";
+const char kDoHProbeHostname[] = "www.ripe.net";
 
 const net::BackoffEntry::Policy kProbeBackoffPolicy = {
     // Apply exponential backoff rules after the first error.
-- 
2.17.1

