From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 26 Nov 2017 11:40:05 +0100
Subject: Restore classic new tab page

---
 chrome/browser/search/search.cc | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/chrome/browser/search/search.cc b/chrome/browser/search/search.cc
--- a/chrome/browser/search/search.cc
+++ b/chrome/browser/search/search.cc
@@ -154,16 +154,6 @@ bool IsURLAllowedForSupervisedUser(const GURL& url, Profile* profile) {
   return true;
 }
 
-bool ShouldShowLocalNewTab(Profile* profile) {
-#if !defined(OS_ANDROID)
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  return command_line->HasSwitch(switches::kForceLocalNtp) ||
-         (base::FeatureList::IsEnabled(features::kUseGoogleLocalNtp) &&
-          profile && DefaultSearchProviderIsGoogle(profile));
-#endif
-  return false;
-}
-
 bool ShouldDelayRemoteNTP(const GURL& search_provider_url, Profile* profile) {
 #if defined(OS_CHROMEOS)
   // On Chrome OS, if the session hasn't merged yet, we need to avoid loading
@@ -190,8 +180,7 @@ struct NewTabURLDetails {
 
     const GURL local_url(chrome::kChromeSearchLocalNtpUrl);
 
-    if (ShouldShowLocalNewTab(profile))
-      return NewTabURLDetails(local_url, NEW_TAB_URL_VALID);
+    return NewTabURLDetails(local_url, NEW_TAB_URL_VALID);
 
     const TemplateURL* template_url =
         GetDefaultSearchProviderTemplateURL(profile);
-- 
2.11.0

