From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 26 Nov 2017 11:40:05 +0100
Subject: Restore classic new tab page

---
 chrome/browser/search/search.cc | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/chrome/browser/search/search.cc b/chrome/browser/search/search.cc
--- a/chrome/browser/search/search.cc
+++ b/chrome/browser/search/search.cc
@@ -156,10 +156,7 @@ bool IsURLAllowedForSupervisedUser(const GURL& url, Profile* profile) {
 
 bool ShouldShowLocalNewTab(Profile* profile) {
 #if !defined(OS_ANDROID)
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  return command_line->HasSwitch(switches::kForceLocalNtp) ||
-         (base::FeatureList::IsEnabled(features::kUseGoogleLocalNtp) &&
-          profile && DefaultSearchProviderIsGoogle(profile));
+  return true;
 #endif
   return false;
 }
-- 
2.11.0

