From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 26 Nov 2017 11:40:05 +0100
Subject: Restore classic new tab page

---
 chrome/browser/search/search.cc | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/chrome/browser/search/search.cc b/chrome/browser/search/search.cc
--- a/chrome/browser/search/search.cc
+++ b/chrome/browser/search/search.cc
@@ -156,13 +156,6 @@ bool IsURLAllowedForSupervisedUser(const GURL& url, Profile* profile) {
   return true;
 }
 
-bool ShouldShowLocalNewTab(Profile* profile) {
-#if !defined(OS_ANDROID)
-  return DefaultSearchProviderIsGoogle(profile);
-#endif
-  return false;
-}
-
 // Used to look up the URL to use for the New Tab page. Also tracks how we
 // arrived at that URL so it can be logged with UMA.
 struct NewTabURLDetails {
@@ -182,8 +175,7 @@ struct NewTabURLDetails {
                              : chrome::kChromeSearchLocalNtpUrl);
 #endif
 
-    if (ShouldShowLocalNewTab(profile))
-      return NewTabURLDetails(local_url, NEW_TAB_URL_VALID);
+    return NewTabURLDetails(local_url, NEW_TAB_URL_VALID);
 
     const TemplateURL* template_url =
         GetDefaultSearchProviderTemplateURL(profile);
-- 
2.17.1

