From: Daniel Micay <danielmicay@gmail.com>
Date: Fri, 28 Jun 2019 17:48:49 -0400
Subject: show download prompt again by default

---
 .../chrome/browser/download/DownloadLocationCustomView.java   | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadLocationCustomView.java b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadLocationCustomView.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadLocationCustomView.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadLocationCustomView.java
@@ -55,10 +55,6 @@ public class DownloadLocationCustomView
     void initialize(@DownloadLocationDialogType int dialogType, File suggestedPath) {
         mDialogType = dialogType;
 
-        // Automatically check "don't show again" the first time the user is seeing the dialog.
-        boolean isInitial =
-                DownloadUtils.getPromptForDownloadAndroid() == DownloadPromptStatus.SHOW_INITIAL;
-        mDontShowAgain.setChecked(isInitial);
         mDontShowAgain.setOnCheckedChangeListener(this);
 
         mFileName.setText(suggestedPath.getName());
-- 
2.17.1

