From bc266421a13f4967a3d5b49830c338f59e4f2dd5 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 1 Aug 2017 20:29:56 -0400
Subject: [PATCH 19/21] stop enabling search engine geolocation by default

---
 .../search_geolocation_service.cc                  | 25 ++++++++++------------
 1 file changed, 11 insertions(+), 14 deletions(-)

diff --git a/chrome/browser/android/search_geolocation/search_geolocation_service.cc b/chrome/browser/android/search_geolocation/search_geolocation_service.cc
index 0ed5540c3262..e5721d53c344 100644
--- a/chrome/browser/android/search_geolocation/search_geolocation_service.cc
+++ b/chrome/browser/android/search_geolocation/search_geolocation_service.cc
@@ -208,7 +208,14 @@ void SearchGeolocationService::SetDSEGeolocationSetting(bool setting) {
   pref.setting = setting;
   SetDSEGeolocationPref(pref);
 
-  ResetContentSetting();
+  url::Origin origin = delegate_->GetDSEOrigin();
+  if (setting) {
+      host_content_settings_map_->SetContentSettingDefaultScope(
+          origin.GetURL(), origin.GetURL(), CONTENT_SETTINGS_TYPE_GEOLOCATION,
+          std::string(), CONTENT_SETTING_ALLOW);
+  } else {
+      ResetContentSetting();
+  }
 }
 
 url::Origin SearchGeolocationService::GetDSEOriginIfEnabled() {
@@ -232,11 +239,7 @@ void SearchGeolocationService::OnDSEChanged() {
   // Remove any geolocation embargo on the URL.
   PermissionDecisionAutoBlocker::GetForProfile(profile_)->RemoveEmbargoByUrl(
       delegate_->GetDSEOrigin().GetURL(), CONTENT_SETTINGS_TYPE_GEOLOCATION);
-  if (content_setting == CONTENT_SETTING_BLOCK && pref.setting) {
-    pref.setting = false;
-  } else if (content_setting == CONTENT_SETTING_ALLOW && !pref.setting) {
-    ResetContentSetting();
-  }
+  pref.setting = content_setting == CONTENT_SETTING_ALLOW;
 
   if (new_dse_name != pref.dse_name && pref.setting)
     SearchGeolocationDisclosureTabHelper::ResetDisclosure(profile_);
@@ -279,7 +282,7 @@ void SearchGeolocationService::InitializeDSEGeolocationSettingIfNeeded() {
 
     PrefValue pref;
     pref.dse_name = delegate_->GetDSEName();
-    pref.setting = content_setting != CONTENT_SETTING_BLOCK;
+    pref.setting = content_setting == CONTENT_SETTING_ALLOW;
     SetDSEGeolocationPref(pref);
 
     SearchGeolocationDisclosureTabHelper::ResetDisclosure(profile_);
@@ -289,13 +292,7 @@ void SearchGeolocationService::InitializeDSEGeolocationSettingIfNeeded() {
 void SearchGeolocationService::EnsureDSEGeolocationSettingIsValid() {
   PrefValue pref = GetDSEGeolocationPref();
   ContentSetting content_setting = GetCurrentContentSetting();
-  bool new_setting = pref.setting;
-
-  if (pref.setting && content_setting == CONTENT_SETTING_BLOCK) {
-    new_setting = false;
-  } else if (!pref.setting && content_setting == CONTENT_SETTING_ALLOW) {
-    new_setting = true;
-  }
+  bool new_setting = content_setting == CONTENT_SETTING_ALLOW;
 
   if (pref.setting != new_setting) {
     pref.setting = new_setting;
-- 
2.14.2

