commit e1675be76a5ab56335b878dcef50043159efae7a
Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Thu Oct 12 08:15:17 2017 +0200

    Disable NTP remote suggestions by default

diff --git a/components/ntp_snippets/features.cc b/components/ntp_snippets/features.cc
index 31823bbc..82eefe3 100644
--- a/components/ntp_snippets/features.cc
+++ b/components/ntp_snippets/features.cc
@@ -38,16 +38,16 @@ const base::Feature* const kAllFeatures[] = {
     nullptr};
 
 const base::Feature kArticleSuggestionsFeature{
-    "NTPArticleSuggestions", base::FEATURE_ENABLED_BY_DEFAULT};
+    "NTPArticleSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kBookmarkSuggestionsFeature{
-    "NTPBookmarkSuggestions", base::FEATURE_ENABLED_BY_DEFAULT};
+    "NTPBookmarkSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kRecentOfflineTabSuggestionsFeature{
     "NTPOfflinePageSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kIncreasedVisibility{"NTPSnippetsIncreasedVisibility",
-                                         base::FEATURE_ENABLED_BY_DEFAULT};
+                                         base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kPhysicalWebPageSuggestionsFeature{
     "NTPPhysicalWebPageSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
