commit 835aa728bc2eaa7f01dc875d87aa64b61bb11ccd
Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Thu Oct 12 01:35:53 2017 +0200

    Disable updater pings

diff --git a/components/component_updater/configurator_impl.cc b/components/component_updater/configurator_impl.cc
index efc46c5..2bb49f36 100644
--- a/components/component_updater/configurator_impl.cc
+++ b/components/component_updater/configurator_impl.cc
@@ -56,8 +56,10 @@ const char kSwitchDisableDeltaUpdates[] = "disable-delta-updates";
 const char kSwitchDisableBackgroundDownloads[] = "disable-background-downloads";
 #endif  // defined(OS_WIN)
 
+#if 0
 const base::Feature kAlternateComponentUrls{"AlternateComponentUrls",
                                             base::FEATURE_ENABLED_BY_DEFAULT};
+#endif
 
 // If there is an element of |vec| of the form |test|=.*, returns the right-
 // hand side of that assignment. Otherwise, returns an empty string.
@@ -89,8 +91,7 @@ ConfiguratorImpl::ConfiguratorImpl(
       fast_update_(false),
       pings_enabled_(false),
       deltas_enabled_(false),
-      background_downloads_enabled_(false),
-      require_encryption_(require_encryption) {
+      background_downloads_enabled_(false) {
   // Parse comma-delimited debug flags.
   std::vector<std::string> switch_values = base::SplitString(
       cmdline->GetSwitchValueASCII(switches::kComponentUpdater), ",",
@@ -138,22 +139,6 @@ int ConfiguratorImpl::UpdateDelay() const {
 
 std::vector<GURL> ConfiguratorImpl::UpdateUrl() const {
   std::vector<GURL> urls;
-  if (url_source_override_.is_valid()) {
-    urls.push_back(GURL(url_source_override_));
-    return urls;
-  }
-
-  if (base::FeatureList::IsEnabled(kAlternateComponentUrls)) {
-    urls.push_back(GURL(kUpdaterDefaultUrlAlt));
-    urls.push_back(GURL(kUpdaterFallbackUrlAlt));
-  } else {
-    urls.push_back(GURL(kUpdaterDefaultUrl));
-    urls.push_back(GURL(kUpdaterFallbackUrl));
-  }
-
-  if (require_encryption_)
-    update_client::RemoveUnsecureUrls(&urls);
-
   return urls;
 }
 
diff --git a/components/component_updater/configurator_impl.h b/components/component_updater/configurator_impl.h
index afc3d2d..5258e47 100644
--- a/components/component_updater/configurator_impl.h
+++ b/components/component_updater/configurator_impl.h
@@ -98,7 +98,6 @@ class ConfiguratorImpl {
   bool pings_enabled_;
   bool deltas_enabled_;
   bool background_downloads_enabled_;
-  bool require_encryption_;
 
   DISALLOW_COPY_AND_ASSIGN(ConfiguratorImpl);
 };
